editor/gl/program.ts

```diff
@@ -58,7 +65,9 @@ export class Program {
         for (let i = 0; i < nb_uniforms; ++i) {
             let uniform_info = gl.getActiveUniform(program, i);
             if (uniform_info) {
-                let loc = gl.getUniformLocation(program, uniform_info.name);
+                // This can't be null as the uniform_info is provided by
+                // the gl context. So we assert that.
```

Comment:
Oh I guess the message is not helping much. X) The cast here (`as ...`) is always valid because be got the uniform_info from the API itself. But `gl.getUniformLocation` does not know that (it doesn't have an appropriate overload) so the returns type contains the `undefined` variant again. Here there's no way we would get a `undefined` `loc` so that was the cast is appropriate.